### PR TITLE
Get file size from S3 after upload + fix upload

### LINKED ISSUE #4944: When uploading the file, the checksum and the size should be retrieved from a head request
**Motivation**
So far, the size is computed within the stream, getting it from a head request would make things easier

**Acceptance criteria**
1. The size is not computed anymore by the local stream

### PR DESCRIPTION
**Pull request by imsdu**
Fixes #4944

### CODE REVIEW

File: delta/kernel/src/main/scala/ch/epfl/bluebrain/nexus/delta/kernel/Lens.scala [hunk 1]
```diff
@@ -1,18 +0,0 @@
-package ch.epfl.bluebrain.nexus.delta.kernel
-
-/**
-  * Allows to resolve an ''S'' to an ''A''
-  */
-trait Lens[S, A] {
```
Comment (imsdu) [inline delta/kernel/src/main/scala/ch/epfl/bluebrain/nexus/delta/kernel/Lens.scala:6]: Dead code is dead

File: delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/files/model/Digest.scala [hunk 2]
```diff
@@ -21,19 +21,23 @@ object Digest {
...
-  }
+  final case class ComputedDigest(algorithm: DigestAlgorithm, value: String) extends Digest
 
   /**
     * A digest that does not yield a value because it is still being computed
     */
-  final case object NotComputedDigest extends Digest {
-    override val computed: Boolean = false
-  }
+  final case object NoDigest extends Digest
```
Comment (imsdu) [inline delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/files/model/Digest.scala:29]: This is slightly different from the other one where we were expecting it to be computed in the future

File: delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/operations/StorageFileRejection.scala [hunk 3]
```diff
@@ -138,6 +137,14 @@ object StorageFileRejection {
           s"File cannot be saved because it already exists on path '$path'."
         )
 
+    /**
+      * Rejection returned when a file can not be saved because content-length is not provided
+      */
+    final case object ContentLengthIsMissing
```
Comment (imsdu) [inline delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/operations/StorageFileRejection.scala:143]: When doing a put with a stream and asking for a checksum, it seems that the content length is needed

File: delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/operations/s3/HeadObject.scala [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.files.model.Digest.ComputedDigest
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.storages.model.DigestAlgorithm
+import software.amazon.awssdk.services.s3.model.HeadObjectResponse
+
+import java.util.Base64
+
+case class HeadObject(fileSize: Long, contentType: Option[ContentType], digest: Digest)
+
+object HeadObject {
+  def apply(response: HeadObjectResponse): HeadObject = {
```
Comment (imsdu) [inline delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/operations/s3/HeadObject.scala:15]: Smart constructor to add more value to the class

**Review by olivergrabinski — APPROVED** [review]

File: tests/docker/config/delta-postgres.conf [hunk 5]
```diff
@@ -1,3 +1,16 @@
+akka {
+  http {
+    client {
+      parsing {
+        max-content-length="100g"
```
Comment (imsdu) [inline tests/docker/config/delta-postgres.conf:5]: To test with larger files

File: tests/src/test/scala/ch/epfl/bluebrain/nexus/tests/HttpClient.scala [hunk 6]
```diff
@@ -119,15 +119,19 @@ class HttpClient private (baseUrl: Uri, httpExt: HttpExt)(implicit
...
-  def uploadFile[A](
-      requestPath: String,
-      fileContents: String,
-      contentType: ContentType,
-      fileName: String,
-      identity: Identity,
-      extraHeaders: Seq[HttpHeader] = jsonHeaders
-  )(assertResponse: (A, HttpResponse) => Assertion)(implicit um: FromEntityUnmarshaller[A]): IO[Assertion] = {
-    def buildClue(a: A, response: HttpResponse) =
+  def uploadFile(project: String, storage: String, file: FileInput, rev: Option[Int])(
```
Comment (imsdu) [inline tests/src/test/scala/ch/epfl/bluebrain/nexus/tests/HttpClient.scala:122]: I refactored the tests to do a general use of `FileInput`

File: tests/src/test/scala/ch/epfl/bluebrain/nexus/tests/kg/files/S3StorageSpec.scala [hunk 7]
```diff
@@ -309,4 +311,27 @@ class S3StorageSpec extends StorageSpec {
       } yield assertion
     }
   }
+
+  "Uploading a large file" should {
+    "succeed" ignore {
```
Comment (imsdu) [inline tests/src/test/scala/ch/epfl/bluebrain/nexus/tests/kg/files/S3StorageSpec.scala:316]: Ignored for now as it takes ~2min to complete
We may want to introduce a slow tag for those and still run them from time to time in a scheduled fashion
(there is at least another one for archives iirc)

**Pull request merged**